server/src/controllers/apps.controller.ts

```diff
@@ -288,4 +285,18 @@ export class AppsController {
     const appUser = await this.appsService.updateVersion(req.user, version, definition);
     return decamelizeKeys(appUser);
   }
+
+  @UseGuards(JwtAuthGuard)
+  @Put(':id/icons')
+  async updateIcon(@Request() req, @Param() params, @Body('icon') icon) {
```

Comment:
Can't we do this update in `PUT /api/apps/:id`? Do we need to add a separate endpoint for this?

Reply:
Since there is a plan to expand this feature, and enable custom icons which user can upload, I think its better to go with separate end point

Reply:
Alright then